src/app/withdraw/page.tsx

```diff
@@ -48,26 +48,39 @@ const Withdraw: React.FC = () => {
 
     const widthDrawFunc = async () => {
         if (!toAddress || !address || !amount) return
-        const amountInShannon = BI.from(amount).mul(BI.from(10).pow(8));
+        const amountInShannon = BI.from(parseFloat(amount) * 10 ** 8)
```

Comment:
Maybe we need to consider checking and handling the potential returns of parseFloat, as parseFloat might return Infinity, I’m not sure how BI.from will deal with such situations.

see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat#:~:text=parseFloat()%20can%20also%20parse%20and%20return%20Infinity%20or%20%2DInfinity%20if%20the%20string%20starts%20with%20%22Infinity%22%20or%20%22%2DInfinity%22%20preceded%20by%20none%20or%20more%20white%20spaces.

Reply:
I did a test, and Bignumber indeed cannot handle infinite. I added a check, so now if `parseFloat(amount)` is Infinite, it will throw an error indicating the amount is invalid.